R/merge.R

```diff
@@ -192,16 +196,22 @@ setGeneric("mergeCols",
     col_data <- colData(x)[element_pos,,drop=FALSE]
     # merge assays
     assays <- assays(x)
-    assays <- S4Vectors::SimpleList(lapply(assays,
-                                           scuttle::sumCountsAcrossCells,
-                                           ids = f,
-                                           subset_row = NULL,
-                                           subset_col = NULL,
-                                           ...))
+    assays <- S4Vectors::SimpleList(lapply(assays, FUN = function(mat, ...){
+        temp <- scuttle::summarizeAssayByGroup(mat,
+                                               ids = f,
+                                               subset.row = NULL,
+                                               subset.col = NULL,
+                                               ...)
+        # "sum" includes agglomerated (summed up) data
+        mat <- assay(temp, "sum")
+        return(mat)
+    }))
```

Comment:
Just call call `scuttle::summarizeAssayByGroup` directly. I think the lapply is not necessary anymore. Just add the assays from the result to `x` in line 212

Reply:
![image](https://user-images.githubusercontent.com/60338854/137583812-a297365d-625b-47f3-a018-10de90798beb.png)

I think that is not possible. 

- Input can be matrix or SE
- Output is SE that includes assays that are calculated from matrix or assay(SE, abund_values). --> Only one matrix would be then agglomerated

https://rdrr.io/github/LTLA/scuttle/man/summarizeAssayByGroup.html

Reply:
Here is me being wrong 😄 

Can you have a look at `.merge_rows` so that it matches this implementation?

Reply:
Now I have also modified `.merge.rows`